b2sdk/file_version.py

```diff
@@ -62,8 +62,11 @@ def __init__(
         action,
         content_md5=None,
         server_side_encryption: Optional[EncryptionSetting] = None,  # TODO: make it mandatory in v2
-        legal_hold: Optional[bool] = None,
-        file_retention: Optional[FileRetentionSetting] = None,
+        file_retention: Optional[
+            FileRetentionSetting
+        ] = None,  # TODO: in v2 change the default value to NO_RETENTION_FILE_SETTING
+        legal_hold: Optional[LegalHold
```

Comment:
the logic for actual value would be very "spreaded" - either `legal_hold` is not None and OFF or UNSET or it is actually None... can we make it not Optional right now and have consistent logic?
same with FileRetentionSetting - why we want to wait till v2?

Reply:
I thought It would be weird for SDK users to not set it (legal_hold or file_retention) and have it set to some value that's meaningful. All possible values of legal_hold (ON, OFF, UNKNOWN, UNSET) carry some meaning and placing them in user created FileVersion objects seems like a violation. 

But I don't know, honestly. Maybe no one constructs these objects. Maybe it's not a violation. What do you think? In v2 I'd make it a mandatory argument and non-nullable.